cogs/weightcog.py

```diff
@@ -557,6 +560,69 @@ async def export(self, ctx: commands.Context, user: discord.Member = None) -> No
                 file=discord.File(csv_file, filename=f"{user.display_name}_weight_data.csv"),
             )
 
+    @commands.command()
+    async def delete_all_user_data(
+        self, ctx: commands.Context, user: discord.Member = None
+    ) -> None:
+        """
+        Deletes all log data related to weight only (CSV file) associated with a
+        specified Discord user or the command invoker.
+
+        Parameters:
+            ctx (commands.Context): The context of the command.
+            user (discord.Member, optional): The Discord user for whom to delete the log data. Defaults to the command invoker.
+        Example:
+            .del_all_log
+            .del_all_log user(Optional)
+        """
+        if user and (user != ctx.author) and not has_bot_input_perms(ctx):
+            await ctx.send(
+                "You don't have the bot-input role and are therefore not allowed to specify"
+                " other users."
+            )
+            return
+
+        user = user or ctx.author
+        user_id = str(user.id)
+        file_path = os.path.join(self.data_folder, f"{user_id}.csv")
+
+        # if the file/data for user does not exist
+        if not os.path.exists(file_path):
+            await ctx.send("No weight data found for this user.")
+            return
+        else:
+            embed = discord.Embed(
+                title=f"Are you sure you want to remove all user weight data for @{user}",
+                timestamp=datetime.utcnow(),
+                colour=discord.Colour.blurple(),
+            )
+            embed.set_thumbnail(url=user.avatar)
+            msg = await ctx.send(embed=embed)
+            for emoji in ["✅", "❌"]:
```

Comment:
I think that emoji's in code will be pretty annoying to maintain or parse in some cases. Could we use unicode? Or the discord naming of emoji's?

:white_check_mark: and :x:

Reply:
Discord naming of emojis doesn't work. The correct way is to use the emoji ID however it only works for custom emojis and not default emojis.
With Unicode, the if condition does not read the unicode value.
Also, there should not be any problem in parsing in the future as I've seen this way only till now in all bots I've worked on.